### ISSUE TITLE
Recommend using "separation of concerns" for viewing lists of posts and whether they are public or draft

### ISSUE THREAD

**Issue by Eilon** [VenusInterns/BlogTemplate#48]
Currently the Index page and Drafts page both have the same "model" behavior: the model retrieves *all* blog posts, regardless of their visibility state (public or draft).

Then the view pages have their own logic for filtering the posts: when they enumerate the full list of posts, they apply some (simple) filtering logic to determine which subset of posts to show.

Instead of that, the views should not know what kind of posts they are showing: they should be provided with an already-filtered list of posts, and then they can iterate over them. You want "smart" models and "dumb" views in general.

Here are the "offending" lines:
* Retrieving the data without filtering:
   * https://github.com/VenusInterns/BlogTemplate/blob/master/BlogTemplate/Pages/Index.cshtml.cs#L30
   * https://github.com/VenusInterns/BlogTemplate/blob/master/BlogTemplate/Pages/Drafts.cshtml.cs#L27
* Listing the data and applying filtering:
   * https://github.com/VenusInterns/BlogTemplate/blob/master/BlogTemplate/Pages/Index.cshtml#L6-L17
   * https://github.com/VenusInterns/BlogTemplate/blob/master/BlogTemplate/Pages/Drafts.cshtml#L11-L20

I recommend that instead of the models doing this:

```c#
            _blog.Posts = _dataStore.GetAllPosts();
```

That they do something like this:

```c#
            _blog.PublicPosts = _dataStore.GetAllPosts().Where(post => post.IsPublic);
```

And then the views get "dumber" and instead of this:

```c#
    foreach (var post in Model.Posts)
    {
        if (!post.IsPublic)
        {
```

Do something like this:

```c#
    foreach (var post in Model.PublicPosts)
    {
```

**Comment by lucasisaza** [comment]
This issue has been addressed, see PR.

**Comment by lucasisaza** [comment]
Fixed this issue with #57
